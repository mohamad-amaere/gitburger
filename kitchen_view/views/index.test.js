const fs = require("fs");
const path = require("path");
const ejs = require("ejs");
const { JSDOM } = require("jsdom");

describe("index.ejs", () => {
	let document;

	beforeAll(async () => {
		const filePath = path.join(__dirname, "index.ejs");
		const template = fs.readFileSync(filePath, "utf-8");
		const orders = [
			{
				orderId: "123",
				orderDate: "2023-10-01",
				items: [
					{
						item: { burger_name: "Cheeseburger" },
						quantity: 2,
						ingredients: [
							{ customization: "Extra", name: "Cheese" },
							{ customization: "No", name: "Onions" }
						]
					}
				]
			}
		];
		const html = await ejs.render(template, { orders }, { async: true });

		const dom = new JSDOM(html);
		document = dom.window.document;
	});

	it("should match the snapshot", () => {
		expect(document.documentElement.outerHTML).toMatchSnapshot();
	});
});
