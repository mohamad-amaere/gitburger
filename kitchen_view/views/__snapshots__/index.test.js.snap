// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.ejs should match the snapshot 1`] = `
"<html lang="en"><head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <h1>Kitchen view</h1>
  
    <div>
      <h2>Order ID: 123</h2>
      <p>Order Date: 2023-10-01</p>
      <h2>Items</h2>
      <ul>
        
          <li>Cheeseburger - 2</li>

          <h4>Ingredient customization</h4>
          
            <li>ExtraCheese</li>
          
            <li>NoOnions</li>
          
        

      </ul>
    </div>
  

</body></html>"
`;
